target/scripts/helpers/ssl.sh

```diff
@@ -408,6 +380,54 @@ function _setup_ssl
   esac
 }
 
+
+# Identify a valid letsencrypt FQDN folder to use.
+function _find_letsencrypt_domain
+{
+  local LETSENCRYPT_DOMAIN
+
+  if [[ -n ${SSL_DOMAIN} ]] && [[ -e /etc/letsencrypt/live/$(_strip_wildcard_prefix "${SSL_DOMAIN}")/fullchain.pem ]]
+  then
+    LETSENCRYPT_DOMAIN=$(_strip_wildcard_prefix "${SSL_DOMAIN}")
+  elif [[ -e /etc/letsencrypt/live/${HOSTNAME}/fullchain.pem ]]
+  then
+    LETSENCRYPT_DOMAIN=${HOSTNAME}
+  elif [[ -e /etc/letsencrypt/live/${DOMAINNAME}/fullchain.pem ]]
+  then
+    LETSENCRYPT_DOMAIN=${DOMAINNAME}
+  else
+    _log 'warn' "Cannot find a valid DOMAIN for '/etc/letsencrypt/live/<DOMAIN>/', tried: '${SSL_DOMAIN}', '${HOSTNAME}', '${DOMAINNAME}'"
+    dms_panic__misconfigured 'LETSENCRYPT_DOMAIN' '_find_letsencrypt_domain'
+  fi
+
+  return "${LETSENCRYPT_DOMAIN}"
```

Comment:
I am not sure what is happening here 😆 I know exactly what you wanted to do though:

``` SUGGESTION
  printf '%s' "${LETSENCRYPT_DOMAIN}"
```

Reply:
Yeah. `return` is only for exit codes 👍

Any reason why you suggest `printf '%s'` over `echo` here?

Reply:
I'm always afraid of `echo`'s newline at the end. This has caused me tremendous trouble in the past (with Kubernetes secrets that contained a newline when they shouldn't), so I generally prefer `printf`. I guess there is not much difference in this case, and we could as well use `echo -n` - I don't really care, as long as the newline can be avoided :)

Reply:
Get it, but I don't think there is a need to avoid the newline. Bash handles this just fine, afaik. This is done in a lot of places already in DMS.

Reply:
Yes, I'm just used to `printf`, that's why I proposed it here as well. If we want to use `echo` here, that's absolutely fine with me.

Reply:
I had a look at the remaining file and it uses `echo`'s too. Otherwise I would be fine with `printf` too.

Reply:
> I had a look at the remaining file and it uses `echo`'s too. Otherwise I would be fine with `printf` too.

Then I will adjust my suggestions.

Reply:
> I know exactly what you wanted to do though

Whoops, I haven't written shell scripts for a while :laughing: